### PR TITLE
chore: Extract common swagger setup configuration + summary api swagger/exception handling

### PR DESCRIPTION
**Pull request by Jonathanio123**
- [x] New feature
- [ ] Bug fix
- [ ] High impact

**Work description**:

This adds some standard boilerplate code that fusion apis use. This includes swagger auth setup, exception middleware and trace middleware.

I extracted the swagger setup from resources api and added it to a new project, this means that summary api can also reuse the same swagger setup. For Exception- and trace middleware, I simply copied them as it has been unchanged for some years and resources has its own exception type in the middleware. 

- ~You now need to authenticate to access swagger for both resources and summary~, reverted as this is a public repo
- Updated swagger configuration so that it is now possible to authenticate trough the swagger page

Some swagger code is taken from https://github.com/equinor/fusion-libraries/tree/master/src/InfrastructureInternal/Fusion.Infrastructure.Internal.Services/Swagger

Link to workitem: [AB#54784](https://statoil-proview.visualstudio.com/787035c2-8cf2-4d73-a83e-bb0e6d937eec/_workitems/edit/54784)

**Test description**:

- [x] Can be tested
- [x] Automatic tests created / updated
- [x] Local tests are passing

**Checklist**:

- [x] Considered automated tests
- [x] Considered updating specification / documentation
- [x] Considered work items
- [x] Considered security
- [x] Performed developer testing
- [x] Checklist finalized / ready for review

### CODE REVIEW

File: src/backend/Fusion.Resources.Infrastructure/Swagger/PatchPropertySchemaFilter.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
+﻿using System.Reflection;
+using Fusion.AspNetCore.Api;
+using Microsoft.OpenApi.Models;
+using Swashbuckle.AspNetCore.SwaggerGen;
+
+namespace Microsoft.Extensions.DependencyInjection;
+
+public class PatchPropertySchemaFilter : ISchemaFilter
```
Comment (Jonathanio123) [inline src/backend/Fusion.Resources.Infrastructure/Swagger/PatchPropertySchemaFilter.cs:8]: Taken from https://github.com/equinor/fusion-libraries Fusion.Infrastructure.Internal.Services

File: src/backend/Fusion.Resources.Infrastructure/Swagger/EnumSchemaFilter.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+﻿using Microsoft.OpenApi.Any;
+using Microsoft.OpenApi.Models;
+using Swashbuckle.AspNetCore.SwaggerGen;
+
+namespace Microsoft.Extensions.DependencyInjection;
+
+public class EnumSchemaFilter : ISchemaFilter
```
Comment (Jonathanio123) [inline src/backend/Fusion.Resources.Infrastructure/Swagger/EnumSchemaFilter.cs:7]: Taken from https://github.com/equinor/fusion-libraries Fusion.Infrastructure.Internal.Services

**Comment by larfeq** [comment]
Summary API relies on Fusion.Events.Server (and Client), but do we need them? I think they can be removed and my build errors will go away :-)

**Comment by Jonathanio123** [comment]
I can try to removing them

**Comment by larfeq** [comment]
Bumping Fusion.Events.Server.csproj from 8.0.0 to 8.0.1 resolves a local build error for me

`<PackageReference Include="Fusion.Events.Server" Version="8.0.1" />`

**Comment by Jonathanio123** [comment]
Removed a bunch of unused packages from the shared project

**Review by aleklundeq — APPROVED** [review]

**Pull request merged**